src/NUnitFramework/tests/Constraints/EqualConstraintTests.cs

```diff
@@ -570,6 +570,14 @@ public void UsesProvidedGenericComparison()
             }
 
             [Test]
+            public void UsesProvidedGenericEqualityComparison()
+            {
+                var comparer = new GenericEqualityComparison<int>();
+                Assert.That(2 + 2, Is.EqualTo(4).Using<int>(comparer.Delegate));
+                Assert.That(comparer.WasCalled, "Comparer was not called");
+            }
```

Comment:
Can we add a test that uses a lambda expression as envisioned by the original issue?